### PR TITLE
Oauth2 login with authorization header

### PR DESCRIPTION
**Pull request by MarcialRosales**
This PR updates the management ui section and oauth2 examples in order to introduce a new capability added to the Oauth2 support in the management ui.

This PR is linked to the [PR](https://github.com/rabbitmq/rabbitmq-server/pull/7365) on the rabbitmq-server.

### CODE REVIEW

File: site/management.md [hunk 1]
```diff
@@ -401,24 +402,20 @@ In addition to the `connect-src` CSP header, RabbitMQ also needs the CSP directi
 
 By default, the RabbitMQ Management UI uses the OAuth 2.0 **authorization code flow** to authenticate and authorize users.
 However, there are scenarios where users preferred to be automatically redirected to RabbitMQ without getting
-involved in additional logon flows. This is common in Web Portals where with a single click, users navigate
-straight to a RabbitMQ cluster's management UI with a token obtained under the covers. This is known as
+involved in additional logon flows. This is common in Oauth2 proxies and in Web Portals where with a single click, users navigate straight to a RabbitMQ cluster's management UI with a token obtained under the covers. This is known as
 **Identity-Provider initiated logon**.
```
Comment (pstack2021) [inline site/management.md:407]: Hi @MarcialRosales first sentence in this section (text not in this review), the word "preferred" should be "prefer"

So reading the entire content above are we trying to say here that customers don't want to be involved in additional logon flows and that OAuth2 proxies and web portals help to avoid this - if this is what we are trying to say, I would rephrase the above to: 

_However, there are scenarios where users prefer to be automatically redirected to RabbitMQ without getting involved in additional logon flows. By using OAuth2 proxies and web portals, these additional logon flows can be avoided. With a single click, users navigate straight to a RabbitMQ cluster's management UI with a token obtained under the covers. This is known as **Identity-Provider initiated logon**._



Does "Web Portals" need to be in uppercase, if not I would change it to "web portals" mid sentence 

I think "RabbitMQ cluster's management UI" should be "RabbitMQ cluster management UI", the use of the word "a" before this makes the instance of "RabbitMQ cluster management UI" singular or plural which is what you are doing here

Also should "management UI" be in uppercase "Management UI" when I scroll up I see "Management UI" I guess use one or the other but keep it consistent throughout

File: site/management.md [hunk 2]
```diff
@@ -401,24 +402,20 @@ In addition to the `connect-src` CSP header, RabbitMQ also needs the CSP directi
...
-
-This is the minimum required configuration for a RabbitMQ cluster configured with `idp_initiated` logon type:
-
 <pre class="lang-ini">
 management.oauth_enabled = true
 management.oauth_initiated_logon_type = idp_initiated
 management.oauth_provider_url = https://my-web-portal
 </pre>
 
+With the settings above, the management UI exposes the HTTP endpoint `/login` which accepts `content-type: application/x-www-form-urlencoded` and it expects the JWT token in the `access_token` form field. This is the endpoint where the web portal would redirect users to access the RabbitMQ Management ui.
```
Comment (pstack2021) [inline site/management.md:417]: instead of the word "above" use "previous" try avoiding use words (where possible) like "above", "below", "top", and "bottom" etc in customer docs especially for readers with accessibility needs for example low visual users will not know where "above" is on a page.

File: site/management.md [hunk 3]
```diff
@@ -401,24 +402,20 @@ In addition to the `connect-src` CSP header, RabbitMQ also needs the CSP directi
...
+Additionally, RabbitMQ also accepts a JWT token in the HTTP `Authorization` header when the user lands on the management UI.
```
Comment (pstack2021) [inline site/management.md:419]: can we use the word "accesses" or "get to" or "opens" the Management UI also should we call it the "RabbitMQ Management UI" throughout this doc to be consistent and clear?

File: site/oauth2-examples-proxy.md [hunk 4]
```diff
@@ -0,0 +1,75 @@
+# Use OAuth2 Proxy and Keycloak as OAuth 2.0 server
+
+Let's test the following flow:
+
+* Access management ui via a browser thru OAuth2-Proxy
```
Comment (pstack2021) [inline site/oauth2-examples-proxy.md:5]: ok update the comment to read "Access the RabbitMQ Management UI using a browser through OAuth2-Proxy"

File: site/oauth2-examples-proxy.md [hunk 5]
```diff
@@ -0,0 +1,75 @@
...
+                       |   | 4. token
+        2.redirect     |  \|/                                        [ RabbitMQ ]
+                [ Oauth2-Proxy ]       ----5. forward with token-->  [  http    ]
+                      /|\
+                       |
+            1. rabbit_admin from a browser
+</pre>
+
+## Prerequisites to follow this guide
+
```
Comment (pstack2021) [inline site/oauth2-examples-proxy.md:19]: Should the heading here read "Prerequisites to use OAuth2 Proxy", it isn't really a guide it is a topic with procedures to setup OAuth2 Proxy, also I think updating the heading will help with SEO (retrieving this topic in google searches)

File: site/oauth2-examples-proxy.md [hunk 6]
```diff
@@ -0,0 +1,75 @@
...
+- Docker
+- make
+
+## Deploy Key Cloak
+
+First, deploy **Key Cloak**. It comes preconfigured with all the required scopes, users and clients.
```
Comment (pstack2021) [inline site/oauth2-examples-proxy.md:25]: Change to "Deploy Key Cloak (I don't think Key Cloak needs to be bolded like it is above) by running the following command: 

Then after the command, you can say add a note to say: 

Note: Key Cloak is preconfigured with the required scopes, users, and clients. It is configured with its own signing key and the [rabbitmq.conf](https://github.com/rabbitmq/rabbitmq-oauth2-tutorial/tree/main/conf/oauth2-proxy/rabbitmq.conf) file is also configured with the same signing key.

File: site/oauth2-examples-proxy.md [hunk 7]
```diff
@@ -0,0 +1,75 @@
...
+<pre class="bash">
+make start-keycloak
+</pre>
+
+**Key Cloak** comes configured with its own signing key. And [rabbitmq.conf](https://github.com/rabbitmq/rabbitmq-oauth2-tutorial/tree/main/conf/oauth2-proxy/rabbitmq.conf)
+is also configured with the same signing key.
+
+To access KeyCloak management interface go to http://0.0.0.0:8080/ and enter `admin` as username and password.
```
Comment (pstack2021) [inline site/oauth2-examples-proxy.md:33]: Reword to "To access the KeyCloak Management UI, go to.......

File: site/oauth2-examples-proxy.md [hunk 8]
```diff
@@ -0,0 +1,75 @@
...
+the setting below which disables validation of the audience claim.
+
+<pre class="lang-ini">
+auth_oauth2.verify_aud = false
+</pre>
+
+
+## Start OAuth2 Proxy
+
+To start Oauth2-proxy we run the following command:
```
Comment (pstack2021) [inline site/oauth2-examples-proxy.md:63]: Avoid "we" in customer docs so simply "To start Oauth2-proxy, run the following command:

File: site/oauth2-examples-proxy.md [hunk 9]
```diff
@@ -0,0 +1,75 @@
...
+
+<pre class="lang-plain">
+make start-oauth2-proxy
+</pre>
+
+Oauth2-proxy is configured using [Alpha configuration](https://github.com/rabbitmq/rabbitmq-oauth2-tutorial/tree/main/conf/oauth2-proxy/alpha-config.yaml). This type of configuration permits injecting the access token into the HTTP **Authorization** header.
```
Comment (pstack2021) [inline site/oauth2-examples-proxy.md:69]: Again do they need to know the above? If so change the word "injecting" to "inserting" (if it makes sense)

File: site/oauth2-examples-proxy.md [hunk 10]
```diff
@@ -0,0 +1,75 @@
...
+
+
+## Access Management ui
+
+Go to http://0.0.0.0:4180/, click on the link **Sign in with Keycloak OIDC**, and enter the credentials
```
Comment (pstack2021) [inline site/oauth2-examples-proxy.md:74]: move the word "link" to after **Sign in with Keycloak OIDC**

File: site/oauth2-examples-proxy.md [hunk 11]
```diff
@@ -0,0 +1,75 @@
...
+`rabbit_admin` as username and `rabbit_admin` as password. You should be redirected to RabbitMQ management ui.
```
Comment (pstack2021) [inline site/oauth2-examples-proxy.md:75]: "management ui" should be "Management UI" - keep consistency in docs

File: site/oauth2-examples.md [hunk 12]
```diff
@@ -151,8 +152,18 @@ in `advanced.config`:
 
 ### <a id="identity-provider-initiated-logon" class="anchor" href="#identity-provider-initiated-logon">Identity-Provider initiated logon</a>
 
-When RabbitMQ is provided as a service from a web portal, it is easy to navigate to the RabbitMQ Management UI
-with a single click. The web portal retrieves a token before taking the user to the RabbitMQ Management UI web page.
+Alike Service-Provider initiated logon, with Idp-initiated logon users land to RabbitMQ management ui with a valid token.
```
Comment (pstack2021) [inline site/oauth2-examples.md:155]: - Change "Alike" to "Like"
 - Also, can you change "land on" to "get to"/"access"/"open" (whichever makes more sense)
  - Change "management ui" to "Management UI"

File: site/oauth2-examples.md [hunk 13]
```diff
@@ -151,8 +152,18 @@ in `advanced.config`:
...
+These two scenarios below are examples of Idp-initiated logon:
```
Comment (pstack2021) [inline site/oauth2-examples.md:156]: Change: **These two scenarios below are examples of Idp-initiated logon:** to **The following scenarios are examples of Idp-initiated logon:**

File: site/oauth2-examples.md [hunk 14]
```diff
@@ -151,8 +152,18 @@ in `advanced.config`:
...
+
+* RabbitMQ is behind a web portal which conveniently allow users to navigate directly to RabbitMQ fully authenticated
+* There is an OAuth2 proxy in between users and RabbitMQ which intercepts their requests and forwards them to RabbitMQ injecting the token into the HTTP `Authorization` header
```
Comment (pstack2021) [inline site/oauth2-examples.md:159]: - Maybe change "injecting" to "inserting"
  - Add full stops at the end of the bullet points - they are full sentences

File: site/oauth2-examples.md [hunk 15]
```diff
@@ -151,8 +152,18 @@ in `advanced.config`:
...
-with a single click. The web portal retrieves a token before taking the user to the RabbitMQ Management UI web page.
+Alike Service-Provider initiated logon, with Idp-initiated logon users land to RabbitMQ management ui with a valid token.
+These two scenarios below are examples of Idp-initiated logon:
+
+* RabbitMQ is behind a web portal which conveniently allow users to navigate directly to RabbitMQ fully authenticated
+* There is an OAuth2 proxy in between users and RabbitMQ which intercepts their requests and forwards them to RabbitMQ injecting the token into the HTTP `Authorization` header  
+
+The latter scenario is demonstrated [here](oauth2-examples-proxy.html). The former scenario is covered in the following section.
+
+#### Idp-initiated logon via the login endpoint
```
Comment (pstack2021) [inline site/oauth2-examples.md:163]: Change the heading to uppercase for letters in main words and change "via" to "using" if it makes sense so rewrite of this heading would now be: Idp-initiated Logon using the Login Endpoint

File: site/oauth2-examples.md [hunk 16]
```diff
@@ -151,8 +152,18 @@ in `advanced.config`:
...
+
+A web portal offers their authenticated users, the option to navigate to RabbitMQ by submitting a form with their OAuth  token in `access_token` form field as it is illustrated below:
+
```
Comment (pstack2021) [inline site/oauth2-examples.md:166]: - remove the comma after users in the first sentence. 
  - I think there is an extra space after "OAuth" and "token" in the 2nd sentence - remove it
   - add the word "the" before `access_token`
   - Lets not say "As it is illustrated below" for accessibility users, how about. "...form field as provided in the following example:

File: site/oauth2-examples.md [hunk 17]
```diff
@@ -161,9 +172,7 @@ with a single click. The web portal retrieves a token before taking the user to
       1. rabbit_admin from a browser                                   3. validate token        
 </pre>
 
-How it works, firstly, the `rabbit_admin` user navigates to the web portal and clicks on the hyperlink associated with a RabbitMQ
-cluster. Next, the web portal obtains a token and redirects the user to RabbitMQ `/login` endpoint with the token within the HTTP form field `access_token`. Finally,
-RabbitMQ validates the token in the http request and if it is valid, it redirects the user to the overview page.
+If the access token is valid, RabbitMQ redirects the user to the overview page.
```
Comment (pstack2021) [inline site/oauth2-examples.md:175]: "is "Overview page" the actual name of a page on the "RabbitMQ Management UI" if so bold it and name it exactly as it is on the UI: **Overview** page

**Review by pstack2021 — CHANGES_REQUESTED** [review]
Hi @MarcialRosales 
Review completed.
Thanks and best regards,
Paula

File: site/oauth2-examples-proxy.md (hunk 5 above)
Comment (MarcialRosales) [reply site/oauth2-examples-proxy.md:19]: Hi @pstack2021 , the prerequisites are only to follow the instructions shown on this use case or guide on how to use OAuth2 Proxy and Keycloak with RabbitMQ. Not necessarily to use OAuth2 proxy. We have 3 more guides like this one where we use the same heading. 
Will the main heading, h1, be enough to help with SEO? That was a very nice advice !

File: site/oauth2-examples-azure.md [hunk 18]
```diff
@@ -118,6 +118,13 @@ Now that some roles have been created for your application, you still need to as
 
 9. Repeat the operations for all the roles you want to assign.
 
+## About custom signing keys
```
Comment (pstack2021) [inline site/oauth2-examples-azure.md:121]: How about changing the heading Optional: Create Custom Signing Keys 
(remember all caps on the 1st letters of main words in headings)

File: site/oauth2-examples-azure.md [hunk 19]
```diff
@@ -118,6 +118,13 @@ Now that some roles have been created for your application, you still need to as
...
+
+You do not need to create a custom signing key for your application. If you create one though, you must append an `appid` query parameter containing the *app ID* to the  `jwks_uri`.
```
Comment (pstack2021) [inline site/oauth2-examples-azure.md:123]: How about changing the above to: "It is optional to create a signing key for your application. If you create one, you must......."

File: site/oauth2-examples-azure.md [hunk 20]
```diff
@@ -118,6 +118,13 @@ Now that some roles have been created for your application, you still need to as
...
+
+For example `https://login.microsoftonline.com/{tenant}/.well-known/openid-configuration?appid=<my-app-id>` contains a `jwks_uri` of  `https://login.microsoftonline.com/{tenant}/discovery/keys?appid=<my-app-id>`.
+
```
Comment (pstack2021) [inline site/oauth2-examples-azure.md:126]: Do we need to show an example of what the variable <my-app-id> is here

Comment (MarcialRosales) [reply site/oauth2-examples-azure.md:126]: It is not necessary.

**Comment by michaelklishin** [comment]
@pstack2021 thank you for the feeback. We use "management UI" in dozens of places and "Management UI" only at the beginning of the sentence. We do not capitalize other feature
names either.
